app/gui/view/src/project/project_view_top_bar/go_to_dashboard_button.rs

```diff
@@ -0,0 +1,100 @@
+//! Provides a button to switch back from the project view to the dashboard.
+
+use crate::prelude::*;
+
+use ensogl::application::tooltip;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::style;
+use ensogl_component::toggle_button;
+use ensogl_component::toggle_button::ToggleButton;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// The width and height of the button.
+pub const SIZE: Vector2 = Vector2(12.0, 16.0);
+
+
+
+// ============
+// === Icon ===
+// ============
+
+/// Defines an icon for returning to the dashboard. It looks like a hamburger button.
+mod icon {
+    use super::*;
+
+    use ensogl::data::color;
+    use ensogl_component::toggle_button::ColorableShape;
+
+    ensogl::shape! {
+        (style: Style, color_rgba: Vector4<f32>) {
+            let fill_color = Var::<color::Rgba>::from(color_rgba);
+            let width = Var::<Pixels>::from("input_size.x");
+            let height = Var::<Pixels>::from("input_size.y");
+            let unit = &width / SIZE.x;
+            let mid_bar = Rect((&unit * 12.0, &unit * 3.0)).corners_radius(&unit);
+            let top_bar = mid_bar.translate_y(&unit * -5.0);
+            let bottom_bar = mid_bar.translate_y(&unit * 5.0);
+            let all_bars = top_bar + mid_bar + bottom_bar;
+            let shape = all_bars.fill(fill_color);
+            let hover_area = Rect((&width, &height)).fill(INVISIBLE_HOVER_COLOR);
+            (shape + hover_area).into()
```

Comment:
I think I was writing about it in one of the previous PR reviews, but I don't see my comment opened (maybe it was closed and it's hard to find it now). We have a module which defines a lot of icons - why we are defining this icon here rather than defining it where other app icons are defined?

Reply:
The comment is still [here](https://github.com/enso-org/enso/pull/6474#discussion_r1188351886). Yes, there are a lot of comments by now unfortunately.

If I'm not mistaken, icons across the codebase are generally defined in the place where they are just, just like other code, which makes sense to me.

Reply:
1. I don't fully understand your sentence `icons across the codebase are generally defined in the place where they are just, just like other code, which makes sense to me.` - are you sure it contains all words / things you wanted to include in it?
2. The idea is to put icons in one place, so they can be reused by different parts of the application. This icon can be used in more places than only for the "go to dashboard" button. It can be used as a hamburger menu icon, for example. So, we have somewhere in the codebase a place with a lot of icons registered by name and if we can, let's move this icon there and just use it from our "icons database", ok?

Reply:
Wojciech means this place: https://github.com/enso-org/enso/blob/46418f132d365ae1946710a445ce3a7cf9d52343/app/gui/view/component-browser/component-list-panel/grid/src/entry/icon.rs

However, this place does not contain _all_ icons. For example, icons for nodes are not here. And it has some limitations, most prominently: the icon size is constant, set to 16. I see the dashboard button has a size of 18px, so it does not fit here. Also, the location of the file must change if we want to make it having all icons (currently it's code specific to the Component Browser).

I think we should generalize the code (and put all icons there, e.g. node's icon) before requiring adding icons there from PRs.

Reply:
@farmaazon thanks for the link!

The go to dashboard icon size is 16x16, not 18x18, so it fits there. All of our icons in Figma are 16x16 to keep them consistent :) 

Regarding generalizing that place – I'm normally doing code generalization when new things appear that require generalization. Back then, that was the only place with icons. Now, we are using icons in other places as well, so it's natural to me to generalize it now (if I was the author of this PR, I'd do this generalization). Otherwise, we are just making it harder to generalize in the future.